src/H5Rpublic.h

```diff
@@ -75,7 +75,7 @@ typedef haddr_t hobj_ref_t;
 /**
  * Buffer to store heap ID and index
  *
- * This needs to be large enough to store largest #haddr_t in a worst case
+ * This needs to be large enough to store largest haddr_t in a worst case
```

Comment:
large enough to store the largest possible haddr_t?

What's a worst case machine?

Reply:
This change was only to fix doxygen warning.
Another item for the "fix the documentation" issue to fix in all branches.